pkg/minikube/bootstrapper/kubeadm/versions.go

```diff
@@ -62,24 +63,36 @@ type ComponentExtraArgs struct {
 
 var componentToKubeadmConfigKey = map[string]string{
 	Apiserver:         "apiServerExtraArgs",
-	Scheduler:         "schedulerExtraArgs",
 	ControllerManager: "controllerManagerExtraArgs",
+	Scheduler:         "schedulerExtraArgs",
 }
 
-func NewComponentExtraArgs(opts util.ExtraOptionSlice, version semver.Version) ([]ComponentExtraArgs, error) {
+func NewComponentExtraArgs(opts util.ExtraOptionSlice, version semver.Version, featureGates string) ([]ComponentExtraArgs, error) {
 	var kubeadmExtraArgs []ComponentExtraArgs
 	for _, extraOpt := range opts {
-		kubeadmKey, ok := componentToKubeadmConfigKey[extraOpt.Component]
-		if !ok {
+		if _, ok := componentToKubeadmConfigKey[extraOpt.Component]; !ok {
 			return nil, fmt.Errorf("Unknown component %s.  Valid components and kubeadm config are %v", componentToKubeadmConfigKey, componentToKubeadmConfigKey)
 		}
-		extraConfig, err := ExtraConfigForComponent(extraOpt.Component, opts, version)
+	}
+
+	keys := []string{}
+	for k := range componentToKubeadmConfigKey {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
```

Comment:
Why do we need this sort? Keys should be unique, and it looks like we collect the result sinto another map (extraConfig).

Reply:
The extraConfig is only an intermediary map that eventually gets passed to kubeadmExtraArgs, a slice.

I think I did this because a slice of []ComponentExtraArgs is passed to the template

```golang
type ComponentExtraArgs struct {
	Component string
	Options   map[string]string
}
```

For tests, we want to write the slice in alphabetical order according to components, and then for each component, write the options in alphabetical order.

I agree this whole construction is a bit suspect, so any refactoring ideas would be appreciated.